run-tests.php

```diff
@@ -3060,6 +3060,28 @@ function settings2array(array $settings, &$ini_settings): void
             }
         }
     }
+
+    // When both mysqlnd and either of mysqli or pdo_msqli are compile
+    // as shared extensions, mysqlnd.so needs to be loaded before either
+    // of mysqli or pdo_msqli are loaded, as otherwise they will complain
+    // about missing symbols.
+    if (array_key_exists('extension', $ini_settings) === true) {
+        $extensions = [];
+        $mysql_extension_present = false;
+        foreach ($ini_settings['extension'] as $extension) {
```

Comment:
Might be a bit more straight forward: https://3v4l.org/ODLA2

Reply:
There are apparently others  "Just like how mbstring needs to be loaded before exif,", so will have a look to add them, and for that, the noddy code is easier to understand for me.

Reply:
If you want to ensure proper order, you also should load **pdo** before pdo drivers (**pdo_xxx**)
(exif dep on mbstring is optional, and btw, can't find where used...)

Reply:
> exif dep on mbstring is optional, and btw, can't find where used..

ext/exif calls several `zend_multibyte_*()` functions, and these dispatch to MBString functions, if available (falling back to dummy implementations otherwise).

Anyhow, instead of manually building the list of extension in the proper order, shouldn't we use `array_search()`, and remove the element with the found key? And frankly, I don't think we should fix the extension order of supplied INI directives; it's a different issue regarding the `--EXTENSIONS--` sections, but these could be specified in the required order in the first place; and if there are unresolveable conflicts, we could skip the tests.